src/robot/running/handlers.py

```diff
@@ -378,3 +378,26 @@ def _parse_arguments(self, handler_method):
         parser = JavaArgumentParser(type='Test Library')
         signatures = self._get_signatures(handler_method)
         return parser.parse(signatures, self.library.name)
+
+
+class EmbeddedArgsTemplate(object):
+
+    def __init__(self, embedded, orig_handler):
+        self.orig_handler = orig_handler
+        self._orig_run = orig_handler._run
+        self.orig_handler._run = self._run
+        self.name = orig_handler.name
+        self.embedded_name = embedded.name
+        self.embedded_args = embedded.args
+
+    def matches(self, name):
+        return self.embedded_name.match(name) is not None
+
+    def _run(self, context, args):
+        if args:
+            raise DataError("Positional arguments are not allowed when using embedded arguments")
+        return self._orig_run(context, [self] + list(self.embedded_args))
```

Comment:
I don't actually understand why `[self]` should ever be passed there. I noticed functions used in tests had `self` as the first parameter but that seems just plain wrong.

It also seems old tests only had keywords implemented as functions in a module. Some tests for keywords implemented as methods in a class are needed too.